Variable Re-association Doc Updates (#223)
DevCycleHQ/devcycle-docs: docs/home/feature-management/features-and-variables/variables-and-variations.md

> Each Feature contains its own set of Variables which it manages. **By default, upon creation of a Feature, a Boolean Variable will be created which has the same name as the Feature's key for easier reference.** Depending on the Feature type, the default Variations will be pre-set. The most common of which will be the Variations of "Variation OFF" and "Variation ON", with the boolean Variable being set to false and true, respectively.

Reviewer: should probably change to 

"Each Feature manages its own set of Variables." instead of "Each Feature contains its own set of Variables which it manages."

When a user is "Served" a Variation based on the [Targeting Rules](/home/feature-management/features-and-variables/targeting-users), the Variable Values the user receives on their devices or as an API response will be the values for the served Variation.

A user can add as many Variables as they desire by simply clicking the "Add Variable" button. 

![Add Variable Modal](/april-2023-add-variable-modal.png)

> Give your new Variable a **key** as well as a **type**, as well as its **values** For each of the current Variations.

Reviewer: Give your new Variable a **key**, a **type**, and its **values** for each of the current Variations.

> Give your new Variable a **key** as well as a **type**, as well as its **values** For each of the current Variations.

Reviewer: ```suggestion
Give your new Variable a **key** as well as a **type**, as well as its **values** For each of the current Variations.
```

> Give your new Variable a **key** as well as a **type**, as well as its **values** For each of the current Variations.

Reviewer: ```suggestion
Give your new Variable a **key**, a **type**, and its **values** for each of the current Variations.
```

The unique Variable **key** is what is used to reference the Variable in code. Variables cannot be re-used in multiple existing Features, and thus the keys must be fully unique.

> The Variable **Type** is meant to help team-wide enforcement of that Variable type so there are no type mismatches throughout the various uses of the Variable. Types are not enforced by the DevCycle SDKs, however.

Reviewer: "The Variable **Type** is meant to help team-wide enforcement of that Variable type so there are no type mismatches throughout the various uses of the Variable."

Can be

"The Variable **Type** helps enforce consistent usage across the team to avoid type mismatches in different use cases."

(sorry some of these comments are just general ones for the page and not related to your changes specifically, just think we can touch it up if we're in there)

> The Variable **Type** is meant to help team-wide enforcement of that Variable type so there are no type mismatches throughout the various uses of the Variable. Types are not enforced by the DevCycle SDKs, however.

Reviewer: ```suggestion
The Variable **Type** helps enforce consistent usage across the team to avoid type mismatches in different use cases.
```

To fully delete a Variable you must do so via our [Management API](/management-api/#operation/VariablesController_remove).

### Re-associating a Variable
DevCycle has the ability to re-use existing variables and re-associate them to different features. 

In the Variable Key input field, a drop down will display all **unassociated, unarchived** variables that can be re-associated to your feature while also providing you the option to add a net new variable.

![Add Variable Modal Dropdown](/april-2023-add-variable-dropdown.png)

In the example above, the variables `demo` and `realtime-demo` are not associated with any other feature and can be added to my feature. 

If you select an existing, unassociated variable from the dropdown, the Variable type will be populated with the type of the selected variable and cannot be changed. 

![Add New Variable Variable Selected](/april-2023-add-new-variable-variable-selected-tooltip.png)

> If you input a variable key name with the same key as an existing variable, but this variable is archived. The error below will appear, as you must first unarchive the variable. 

Reviewer 2: What do you think so something like:

```suggestion
If you input a variable key that matches an existing archived variable, the error below will appear, as you must first unarchive the variable.
```

![Add New Variable Archived Variable Error](/april-2023-add-variable-modal-archived-variable-error.png)

> If you wish to use it, click on the hyperlinked "**variable**" text and it will take you directly to the archived variable page for you to unarchive. 

Reviewer: "To use it, click the hyperlinked "**variable**" text, and you'll be directed to the archived variable page where you can unarchive it."

:::info

> If you would like to move a variable between features, the variable must first be removed from the previous feature to fall into an unassociated state. This means that the variable will return back to it's default values for the duration it is not associated to a feature. 

Reviewer: "If you would like to move a variable between features, the variable must first be removed from the previous feature to fall into an unassociated state." 

could be clearer as: 

"If you want to move a variable between features, you must first remove it from the previous feature, making it unassociated."
